netbeans/CheckerFrameworkPlugin/src/org/checkerframework/netbeans/CheckerFrameworkPanel.java

```diff
@@ -0,0 +1,146 @@
+                                                            "checkerstrings.properties",
+                                                            "org.checkerframework.netbeans",
+                                                            false)))
+                            .entrySet();
+        } catch (IOException e) {
+            System.out.println("Failed to load checker strings properties file.");
+            checkerStrings = new HashSet(); //create an empty hash set
+        }
+
+        checkerList = new JCheckBox[checkerStrings.size()];
+        int i = 0;
+        String tmp = editableProperty.get("annotation.processing.processors.list");
+        for (Entry<String, String> e : checkerStrings) {
+            checkerList[i] = new JCheckBox(e.getValue());
+            if (tmp.toString().contains(e.getKey())) {
+                checkerList[i].setSelected(true);
+            }
+            this.add(checkerList[i]);
+            i++;
+        }
+    }
+
+    /**
+     * Helper method to load a properties file
+     *
+     * @param propsFO File object to be loaded
+     * @return EditableProperties class loaded with the properties file
+     * @throws IOException
+     */
+    private static EditableProperties loadProperties(FileObject propsFO) throws IOException {
+        InputStream propsIS = propsFO.getInputStream();
+        EditableProperties props = new EditableProperties(true);
+        try {
+            props.load(propsIS);
+        } finally {
+            propsIS.close();
+        }
+        return props;
+    }
+
+    /**
+     * This method is called by the action listener that runs when the Project Properties window is
+     * closed and will store the selected checkers into the project.properties file.
+     *
+     * @throws IOException
+     */
+    public void store() throws IOException {
+        try {
+            ProjectManager.mutex()
+                    .writeAccess(
+                            new WriteCheckerFrameworkPropertiesAction(
+                                    projectProperties,
+                                    checkerPath,
+                                    checkerQualPath,
+                                    updateSelections()));
+        } catch (MutexException mux) {
+            throw (IOException) mux.getException();
+        }
+    }
+
+    public JCheckBox[] getCheckBoxes() {
+        return checkerList;
+    }
+
+    private String updateSelections() {
+        StringBuilder sel = new StringBuilder();
+        String selectedChecker = "";
+        for (JCheckBox checkBox : checkerList) {
+            if (checkBox.isSelected()) {
+                for (Entry<String, String> entry : checkerStrings) {
+                    if (entry.getValue().equals(checkBox.getText())) {
+                        selectedChecker = entry.getKey();
+                    }
+                }
+                //try to add the item
+                if (sel.toString().isEmpty()) {
```

Comment:
This would be clearer if the common part were extracted out:

```
if (sel.length() != 0) {
  sel.append(',');
}
sel.append(selectedChecker);
```

Note how this also prevents the StringBuffer sel to be converted to a String in each iteration!